app/Database/CourseQueries.hs

```diff
@@ -178,6 +179,16 @@ allCourses = do
       return $ T.unlines codes :: SqlPersistM T.Text
   return $ toResponse response
 
+-- | Retrieves the prerequisites for a course (code) as a string.
+prereqsForCourse :: T.Text -> IO (Either String T.Text)
+prereqsForCourse course = runSqlite databasePath $ do
+    courses :: [Entity Courses] <- selectList [CoursesCode ==. course] []
```

Comment:
Use `selectFirst` instead of `selectList` when you only want one record. (It returns a `Maybe` instead of a list.)